### ISSUE TITLE
Documentation could benefit from a few more examples.

### ISSUE THREAD

**Issue by ivandatasci** [adamgordonbell/csvquote#2]
Documentation does not let me conclude whether or not csvquote is the utility for the task I have in mind.

How to use csvquote (and awk) to filter the rows of a CSV file where column "p_value" has values under 0.05?

Users would appreciate that as it is such a common task.

Thank you.

**Comment by adamgordonbell** [comment]
Hi @ivandatasci,

That is very possible, although its more of an awk question than a csvquote one.

You would do it something like this:
```
> cat foobar.csv | csvquote | awk -F, '$4 <= 0.05 { print }  | csvquote -u
```

**Comment by ivandatasci** [comment]
Hi @adamgordonbell.

Thanks for replying.

I am very much convinced that the documentation needs more clarity. Let me provide some examples myself using other popular csv tools.

Lets start with a toy example
```
$ printf 'name,day,p_value\nJane,Tue,0.02\nJohn,Mon,0.11\nLuis,Fri,0.80'
name,day,p_value
Jane,Tue,0.02
John,Mon,0.11
```

If nothing else, the README would benefit with this toy CVS which comprises more than one line. Your reply to my original message is missing a quote thus producing an error but that is not seen easily because it uses a hypothetical foobar.csv. 

Example tool: csvlook
```
$ printf 'name,day,p_value\nJane,Tue,0.02\nJohn,Mon,0.11\nLuis,Fri,0.80' | csvlook --no-inference
| name | day | p_value |
| ---- | --- | ------- |
| Jane | Tue | 0.02    |
| John | Mon | 0.11    |
| Luis | Fri | 0.80    |
```

Example tool: csvcut
```
$ printf 'name,day,p_value\nJane,Tue,0.02\nJohn,Mon,0.11\nLuis,Fri,0.80' | csvcut -c name,p_value
name,p_value
Jane,0.02
John,0.11
Luis,0.80
```

Example tool: csvgrep
```
$ printf 'name,day,p_value\nJane,Tue,0.02\nJohn,Mon,0.11\nLuis,Fri,0.80' | csvgrep -c name -m 'L'
name,day,p_value
Luis,Fri,0.80
```

Notice how in all examples above the CSV header is treated differently from the CSV body and it is displayed with the output.

Now lets try the example above with AWK:
```
$ printf 'name,day,p_value\nJane,Tue,0.02\nJohn,Mon,0.11\nLuis,Fri,0.80' | awk -F, '{if ($1 ~ /L/ ) print $0}'
Luis,Fri,0.80
```

Notice the loss of the header above.

Lets try with the same with csvquote:
```
$ printf 'name,day,p_value\nJane,Tue,0.02\nJohn,Mon,0.11\nLuis,Fri,0.80' | csvquote | awk -F, '{if ($1 ~ /L/ ) print $0}' | csvquote -u
Luis,Fri,0.80
```

Notice again that the header has been lost despite the use of csvquote.

In summary:
My perception is that csvquote is a tool for quoting in general but it does not protect the header to be passed to awk.
I conclude that csvquote is a great tool but it is not a csv tool to be paired with awk. It is not a CSV-specific tool.
The webpage https://earthly.dev/blog/awk-csv/ did not help me understand the limitations of csvquote.

Please consider adding more examples to help clarify.

Thank you gain.

**Comment by adamgordonbell** [comment]
Hi @ivandatasci,
Thanks for the feedback! I'm happy to accept PRs adding to the read me adding further documentation.

>My perception is that csvquote is a tool for quoting in general but it does not protect the header to be passed to awk.

This is correct CVSquote deals only in quoting and unquoting CSV. This is significant because many useful unix tools like  AWK, cut, head, tail and so on can't handle CSV files with newlines or commas in the records.  

So, if you want to specially handle headers in AWK, then use AWK to specially handle headers. CSVquote just handles the quote part. 

Similarly if you know `cut` you may not need to install `csvcut` just use `cut` piping to csvquote first.
